libnd4j/include/ops/declarable/generic/parity_ops/resize_bicubic.cpp

```diff
@@ -29,52 +29,71 @@ namespace nd4j {
-
-            REQUIRE_TRUE(size->lengthOf() == 2, 0, "resize_bilinear: Resize params is a pair of values, not %i.", size->lengthOf());
-            REQUIRE_TRUE(block.numI() <= 1, 0, "resize_bilinear: Resize params already given by the second param. Int params are expensive.");
-            width = size->e<int>(0);
-            height = size->e<int>(1);
-            auto method = 1; //kResizeBilinear;
-            if (block.numI() == 1) {
-                method = INT_ARG(0);
-            }
-            auto preserveAspectRatio = false;
-            auto antialias = false;
+            auto inRank = image->rankOf();
+            REQUIRE_TRUE(inRank == 3 || inRank == 4, 0, "resize_bicubic: Source tensor should have rank 4, but %i given.", inRank);
+            REQUIRE_TRUE(output->rankOf() == inRank, 0, "resize_bicubic: Source tensor and output should have the same rank, but  %i and %i given.", inRank, output->rankOf());
+            REQUIRE_TRUE(size->rankOf() == 1, size->lengthOf() == 2, 0, "resize_bicubic: Resize params is a pair of values, not %i.", size->lengthOf());
+            REQUIRE_TRUE(block.numI() <= 1, 0, "resize_bicubic: Resize params already given by the second param. Int params are expensive.");
+            width = size->e<int>(1);
+            height = size->e<int>(0);
+            REQUIRE_TRUE(width > 0 , 0, "resize_bicubic: picture width should be positive 32 bit integer, but %i given", width);
+            REQUIRE_TRUE(height > 0 , 0, "resize_bicubic: picture height should be positive 32 bit integer, but %i given", height);
+            //REQUIRE_TRUE(image->sizeAt(1) > 3 && image->sizeAt(2) > 3, 0, "resize_cubic: To use bicubic algorithm need at least 16 pixels as source.");
+            REQUIRE_TRUE(width > 3 && height > 3, 0, "resize_bicubic: To use bicubic algorithm need at least 16 pixels as target.");
+            REQUIRE_TRUE(image->lengthOf() > 0, 0, "resize_bicubic: Only non-zero images allowed to processing.");
+//            auto method = 1; //kResizeBilinear;
+//            if (block.numI() == 1) {
+//                method = INT_ARG(0);
+//            }
+            auto alignCorners = false;
+            auto halfPixelAlign = false;
             if (block.numB() > 0) {
-                preserveAspectRatio = block.getBArguments()->at(0);
+                alignCorners = block.getBArguments()->at(0);
                 if (block.numB()> 1)
-                    antialias = block.getBArguments()->at(1);
+                    halfPixelAlign = block.getBArguments()->at(1);
             }
+            REQUIRE_TRUE(halfPixelAlign == false || halfPixelAlign == true && alignCorners == false, 0, "resize_bicubic: half pixel align can be used only with non-aligned corners");
+
+            auto source = inRank == 4?image->reshape(image->ordering(), {image->sizeAt(0), image->sizeAt(1), image->sizeAt(2), image->sizeAt(3)}):image->reshape(image->ordering(), {1, image->sizeAt(0), image->sizeAt(1), image->sizeAt(2)});
```

Comment:
Why are you making reshape to the same shape, if rank is 4?

Reply:
this case is the simplest to avoid copiing data and get reference instread pointer.